### PR TITLE
fix(1082): [10] change way to get jobs from trigger

### PR DESCRIPTION
**Pull request by yuichi10**
## Context
When startFrom is `~commit`, Screwdriver need to start jobs which start from `~commit:branch` and `~commit`.

## Objective
When startFrom is `~commit`, getJobsFromTrigger try to find job which start from `~commit:branch`.

## References
Related to https://github.com/screwdriver-cd/screwdriver/issues/1082

### CODE REVIEW

File: lib/eventFactory.js [hunk 1]
```diff
@@ -18,17 +18,24 @@ let instance;
  * @return {Array}                                          Array of commit jobs to start
```
Comment (tk3fftk) [inline lib/eventFactory.js:19]: We should need to add `config.branch` to JSDoc.

Comment (yuichi10) [reply lib/eventFactory.js:19]: Thanks I will add that to JSDoc.

File: lib/eventFactory.js [hunk 2]
```diff
@@ -15,20 +15,28 @@ let instance;
  * @param  {Object}   config.pipelineConfig
  * @param  {Object}   config.pipelineConfig.workflowGraph   Object with nodes and edges that represent the order of jobs
  * @param  {String}   config.startFrom                      Startfrom (e.g. ~commit, ~pr, ~sd@123:main, etc)
+ * @param  {String}   consig.branch                         triggered branch name
```
Comment (tk3fftk) [inline lib/eventFactory.js]: consig -> config

Comment (yuichi10) [reply lib/eventFactory.js]: ahhh sorry I will fix...

**Review by tk3fftk — APPROVED** [review]
LGTM

**Review by sakka2 — APPROVED** [review]
LGTM!

File: lib/eventFactory.js [hunk 3]
```diff
@@ -145,17 +154,23 @@ function startBuild(config) {
...
     const { eventConfig, eventId, pipeline, pipelineConfig, changedFiles, webhooks } = config;
     const startFrom = eventConfig.startFrom;
+    let branch = '';
 
     if (!startFrom) {
         return null;
     }
 
-    return pipeline.jobs.then((jobs) => {
+    return pipeline.branch.then((b) => {
```
Comment (tkyi) [inline lib/eventFactory.js]: I think you call `pipeline.branch` and `pipeline.jobs` at the same time using `Promise.all` like:
```
return Promise.all([
                pipeline.branch,
                pipeline.jobs
            ]).then(([branch, jobs]) => {
```

File: lib/eventFactory.js [hunk 4]
```diff
@@ -15,20 +15,28 @@ let instance;
  * @param  {Object}   config.pipelineConfig
  * @param  {Object}   config.pipelineConfig.workflowGraph   Object with nodes and edges that represent the order of jobs
  * @param  {String}   config.startFrom                      Startfrom (e.g. ~commit, ~pr, ~sd@123:main, etc)
+ * @param  {String}   config.branch                         triggered branch name
```
Comment (tkyi) [inline lib/eventFactory.js]: can you sort this comment alphabetically?

File: lib/eventFactory.js (hunk 3 above)
Comment (yuichi10) [reply lib/eventFactory.js]: I will fix. Thank you!!

File: lib/eventFactory.js (hunk 4 above)
Comment (yuichi10) [reply lib/eventFactory.js]: I will fix!

Comment (yuichi10) [reply lib/eventFactory.js]: I also fixed this comment.
https://github.com/screwdriver-cd/models/blob/master/lib/eventFactory.js#L37

**Review by tkyi — APPROVED** [review]
:+1: LGTM

**Pull request merged**